### ISSUE TITLE
Cross Validation

### ISSUE THREAD

**Issue by DSP137** [openconnectome/Statistical-Connectomics-Class#227]
When is necessary to use cross validation and which methods of cross validation are most prevalent in practice?

**Comment by dlee138** [comment]
You basically want to use cross validation to see how accurately your model can predict stuff when given new or unseen data.  Sometimes a model might fit known data but fit it too well because it overfits the data.  With cross validation, you compare the training set with known data with the testing set from unnkown data to see how well the model predicts an independent (unknown) dataset.

**Comment by ajulian3** [comment]
Cross Validation, although powerful, can at times be detrimental to a statistical analysis due to over fitting. Specifically, in my experience I have used training and testing data sets to best implement cross validation. In terms of real world applications, this is helpful when analyzing quantitative or spatial information. For example, in data mining, cross validation can be used to detect probabilistic models for email viruses.

**Comment by whock** [comment]
This isn't necessarily specific to connectomics, but here are some major types of cross-validation I've heard of: 1) leave one out 2) leave k-out 2) k-fold cross-validation

**Comment by mrjiaruiwang** [comment]
Cross validation is considered heavily for medical samples. Yes there are problems and bias, but if your training data set has a very low number of samples (like <100), cross-validation is the best way to validate your classifier.  You essentially partition your data multiple times to generate "training" and "testing" data with your small sample-sized dataset so as to give the impression of classifying on a much larger dataset than you actually have. Make sure to always permute the input data before partitioning to ensure consistency. Note that CV is not really necessary if obtaining data is relatively easy or if you have a lot of samples (>10000).

**Comment by itsthefedora** [comment]
To expand on what Jerry said: cross-validation is the *only* way to evaluate the performance of your classifier.

Think of everything statistically: Say you have a random feature vector X and label L that are jointly distributed. Now, you have a dataset T of n samples from this distribution (Xi, Li), i = 1 ... n. Now, you have an algorithm A that takes T, the data, and produces F, a classifier; that is, A(T) is a *function* that takes in an X and spits out its best guess of the L.

The goal of "building a classifier" is to make it so that this F that we trained does well at classifying *new* samples that were uninvolved in F's training; that is, given some loss functional e, we want to minimize the expectation
E[ e(F(X), L) ]
over (X,L)

How do you *estimate* this error, though? T is, in fact, *random*; hence, F = A(T), the classifier, is also *random*. So, to really assess the performance of our *algorithm* A, we need to actually look at the expectation
E[ e( A(T)(X), L ) ]
over T x (X,L); that is, how well would our classifier perform if we *resampled* the training data from the same distribution? It turns out, your estimate of this is *extremely* biased (in the optimistic direction) when you use samples from T as the (X,L) in computing this expectation, because it breaks the implicit assumption that T and (X,L) are *independent*. Cross-validation is, essentially, just bootstrapping by sub-sampling T many times: you end up with a *distribution* of error statistics computed by training on the sub-samples and validating on the complements, giving you a much better picture of the performance of your *algorithm*, A.

Cross-validation isn't an annoyance; it is, in fact, a necessary consideration when you are trying to accurately gauge the performance of your classifier. Testing on the training data is what we call "wrong".
